[CI] FieldExtractorIT testScaledFloat failing

**Build Scans:**
- [elasticsearch-periodic #6550 / openjdk17_8.16.5_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/jeyvs3mg5jygi)
- [elasticsearch-periodic #6522 / openjdk17_8.18.0_java-fips-matrix-bwc](https://gradle-enterprise.elastic.co/s/2xaml66csfxyq)
- [elasticsearch-periodic #6431 / graalvm-ce17_8.17.3_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/qunfhv3ykt5zy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.16.5#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT" -Dtests.method="testScaledFloat" -Dtests.seed=6CD5641AF5310058 -Dtests.bwc=true -Dtests.locale=wae-CH -Dtests.timezone=America/Inuvik -Druntime.java=17
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testScaledFloat),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.net.SocketTimeoutException: 60’000 milliseconds timeout on connection http-outgoing-3 [ACTIVE]
```

**Issue Reasons:**
- [8.x] 3 failures in test testScaledFloat (0.3% fail rate in 946 executions)
- [8.x] 3 failures in pipeline elasticsearch-periodic (16.7% fail rate in 18 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

````
[2025-02-15T22:36:36,089][INFO ][o.e.x.l.SyntheticSourceIndexSettingsProvider] [v8.17.3-2] unable to create mapper service for index [test_index]
java.lang.AssertionError: soft deletes must be enabled in version 8521000
	at org.elasticsearch.index.IndexSettings.<init>(IndexSettings.java:956) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.indices.IndicesService.createIndexMapperServiceForValidation(IndicesService.java:824) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.xpack.logsdb.SyntheticSourceIndexSettingsProvider.newIndexHasSyntheticSourceUsage(SyntheticSourceIndexSettingsProvider.java:127) ~[?:?]
	at org.elasticsearch.xpack.logsdb.SyntheticSourceIndexSettingsProvider.getAdditionalIndexSettings(SyntheticSourceIndexSettingsProvider.java:89) ~[?:?]
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.aggregateIndexSettings(MetadataCreateIndexService.java:1004) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:609) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:448) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:327) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:574) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1075) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1038) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:245) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1691) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener.run(ActionListener.java:452) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1688) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1283) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener.run(ActionListener.java:452) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1262) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27) ~[elasticsearch-8.17.3-SNAPSHOT.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1575) ~[?:?]
```

I see in [log](https://gradle-enterprise.elastic.co/s/2xaml66csfxyq/tests/task/:x-pack:plugin:esql:qa:server:mixed-cluster:v8.18.0%23javaRestTest/details/org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT/testScaledFloat/1/output)

```
[2025-02-20T03:10:06,143][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-1] fatal error in thread [elasticsearch[test-cluster-1][clusterApplierService#updateTask][T#1]], exiting java.lang.AssertionError: provided source [{"_doc":{"_source":{"mode":"synthetic"},"properties":{"scaled_float_field":{"type":"scaled_float","ignore_malformed":false,"scaling_factor":6.700759582567014E18}}}}] differs from mapping [{"_doc":{"_source":{"mode":"synthetic"},"properties":{"scaled_float_field":{"type":"scaled_float","ignore_malformed":false,"scaling_factor":6.7007595825670144E18}}}}]
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.index.mapper.DocumentMapper.<init>(DocumentMapper.java:75)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.index.mapper.MapperService.newDocumentMapper(MapperService.java:606)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.index.mapper.MapperService.updateMapping(MapperService.java:358)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.index.IndexService.updateMapping(IndexService.java:847)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.indices.cluster.IndicesClusterStateService.createIndicesAndUpdateShards(IndicesClusterStateService.java:595)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.indices.cluster.IndicesClusterStateService.doApplyClusterState(IndicesClusterStateService.java:318)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:272)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:544)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:530)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:503)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:432)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:157)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:977)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:218)
	at org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:184)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
@@ -223,7 +223,7 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
     public void testScaledFloat() throws IOException {
         double value = randomBoolean() ? randomDoubleBetween(-Double.MAX_VALUE, Double.MAX_VALUE, true) : randomFloat();
         // Scale factors less than about 5.6e-309 will result in NaN (due to 1/scaleFactor being infinity)
-        double scalingFactor = randomDoubleBetween(1e-308, Double.MAX_VALUE, false);
+        double scalingFactor = randomDoubleBetween(1e-308, Float.MAX_VALUE, false);
         new Test("scaled_float").expectedType("double")
             .randomIgnoreMalformedUnlessSynthetic()
             .randomDocValuesUnlessSynthetic()

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
@@ -221,9 +221,14 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
     }
 
     public void testScaledFloat() throws IOException {
+        // Running this on 17 when nodes in cluster run JDK >17 triggers an assert due to a mismatch
+        // of results produced by Double#toString for some specific numbers.
+        // See https://github.com/elastic/elasticsearch/issues/122984.
+        assumeTrue("JDK version greater than 17", Runtime.version().feature() > 17);
+
         double value = randomBoolean() ? randomDoubleBetween(-Double.MAX_VALUE, Double.MAX_VALUE, true) : randomFloat();
         // Scale factors less than about 5.6e-309 will result in NaN (due to 1/scaleFactor being infinity)
-        double scalingFactor = randomDoubleBetween(1e-308, Float.MAX_VALUE, false);
+        double scalingFactor = randomDoubleBetween(1e-308, Double.MAX_VALUE, false);
         new Test("scaled_float").expectedType("double")
             .randomIgnoreMalformedUnlessSynthetic()
             .randomDocValuesUnlessSynthetic()

